content/courses/program-security/signer-auth.md

```diff
@@ -497,73 +509,101 @@ pub struct SecureWithdraw<'info> {
+        unauthorizedWallet,
+      ]);
+      throw new Error("Expected transaction to fail, but it succeeded");
+    } catch (error) {
+      expect(error).to.be.an("error");
+      console.log("Error message:", error.message);
     }
-  })
+  });
 })
 ```
 
-Run `anchor test` to see that the transaction will now return a signature
+Run `anchor test` to see that the transaction now returns a signature
 verification error.
 
 ```bash
-Error: Signature verification failed
+signer-authorization
+Error message: Signature verification failed.
+Missing signature for public key [`GprrWv9r8BMxQiWea9MrbCyK7ig7Mj8CcseEbJhDDZXM`].
+    ✔ fails to perform secure withdraw with incorrect signer
 ```
 
-That’s it! This is a fairly simple thing to avoid, but incredibly important.
-Make sure to always think through who should who should be authorizing
-instructions and make sure that each is a signer on the transaction.
+This example shows how important it is to think through who should authorize
+instructions and ensure that each is a signer on the transaction.
 
-If you want to take a look at the final solution code you can find it on the
-`solution` branch of
-[the repository](https://github.com/Unboxed-Software/solana-signer-auth/tree/solution).
+To review the final solution code, you can find it on the
+[`solution` branch of the repository](https://github.com/solana-developers/solana-signer-auth/tree/solution).
 
 ## Challenge
 
-At this point in the course, we hope you've started to work on programs and
-projects outside the labs and Challenges provided in these lessons. For this and
-the remainder of the lessons on security vulnerabilities, the Challenge for each
-lesson will be to audit your own code for the security vulnerability discussed
-in the lesson.
+Now that you've worked through the labs and challenges in this course, it's time
+to apply your knowledge in a practical setting. For this challenge and those
+that follow on security vulnerabilities, audit your own programs for the
+specific vulnerability discussed in each lesson.
+
+### Steps
 
-Alternatively, you can find open source programs to audit. There are plenty of
-programs you can look at. A good start if you don't mind diving into native Rust
-would be the
-[SPL programs](https://github.com/solana-labs/solana-program-library).
+1. **Audit Your Program or Find an Open Source Project**:
 
-So for this lesson, take a look at a program (whether yours or one you've found
-online) and audit it for signer checks. If you find a bug in somebody else's
-program, please alert them! If you find a bug in your own program, be sure to
-patch it right away.
+   - Begin by auditing your own code for missing signer checks, or find an open
+     source Solana program to audit. A great place to start is with the
+     [SPL programs](https://github.com/solana-labs/solana-program-library) if
+     you’re comfortable working with native Rust.
```

Comment:
```suggestion
     source Solana program to audit. A great place to start is with the
     [program examples](https://github.com/solana-developers/program-examples) repo.
```

Reply:
This is a better codebase as it has anchor examples (and could use an audit!).

Reply:
Yes, I updated the link for the same.